lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -38,100 +38,44 @@
-            coordinates.
-        master_coord (String):
-            Coordinate that the other coordinates will be associated with.
-        coordinates (None or List):
-            List of coordinates to be associated with the master_coord.
-
-    Returns:
-        forecast_data (Iris cube):
-            Cube where the the requested coordinates have been added to the
-            cube as auxiliary coordinates and associated with the desired
-            master coordinate.
-
-    Raises:
-        ValueError: If the master coordinate is not present on the cube.
-
-    """
-    if coordinates is None:
-        coordinates = []
-
-    if not cube.coords(master_coord):
-        msg = (
-            "The master coordinate for associating other "
-            "coordinates with is not present: "
-            "master_coord: {}, other coordinates: {}".format(
-                master_coord, coordinates))
-        raise ValueError(msg)
-
-    # If the master_coord is not a dimension coordinate, then the other
-    # coordinates cannot be associated with it.
-    if len(cube.coords(master_coord, dim_coords=True)) > 0:
-        for coord in coordinates:
-            if cube.coords(coord):
-                temp_coord = cube.coord(coord)
-                cube.remove_coord(coord)
-                temp_aux_coord = (
-                    build_coordinate(temp_coord.points,
-                                     bounds=temp_coord.bounds,
-                                     coord_type=AuxCoord,
-                                     template_coord=temp_coord))
-                coord_names = [
-                    coord.standard_name for coord in cube.dim_coords]
-                cube.add_aux_coord(
-                    temp_aux_coord,
-                    data_dims=coord_names.index(master_coord))
-    return cube
-
-
-def _slice_over_coordinate(cubes, coord_to_slice_over):
-    """
-    Function slice over the requested coordinate,
-    promote the sliced coordinate into a dimension coordinate
-    to help concatenation.
-
-    Args:
-        cubes (Iris cubelist or Iris cube):
-            Cubes to be concatenated.
-        coords_to_slice_over (List):
-            Coordinates to be sliced over.
+        cubes (iris.cube.CubeList):
+            List of cubes to check the attributes and revise.
 
-    Returns:
-        Iris CubeList
-            CubeList containing sliced cubes.
+    Kwargs:
+        unmatched (list or None):
+            List of unmatched attributes.  If None, function
+            finds.
+        silent (list or None):
+            List of attributes to remove silently if unmatched.
 
+    Warns:
+        UserWarning:
+            If an unmatched attribute is not on the "silent" list,
+            a warning will be raised.
     """
-    sliced_by_coord_cubelist = iris.cube.CubeList([])
-    if isinstance(cubes, iris.cube.Cube):
-        cubes = iris.cube.CubeList([cubes])
+    if unmatched is None:
+        unmatched = compare_attributes(cubes)
```

Comment:
Perhaps not for this PR, but perhaps `compare_attributes` should be named `find_nonmatching_attributes`, as that seems to be a the aim. Perhaps it would make this function slightly simpler if `compare_attributes` just returned a list of the attribute names that haven't matched e.g. `unmatched_attributes=["history"]`, unless there's a reason to return the whole attribute. What do you think?

Reply:
I think this is a good idea.  I'm not sure I can do it in this ticket though, as `compare_attributes` is called elsewhere in the IMPROVER code.  But perhaps this could be follow-on work?

Reply:
Would you be happy to create a small follow-on ticket to cover this work to review and potentially tidy up the naming and output from the `compare_attributes` function?

Reply:
Done